pkg/registry/apis/dashsnap/register.go

```diff
@@ -0,0 +1,259 @@
+}
+
+// Register additional routes with the server
+func (b *SnapshotsAPIBuilder) GetAPIRoutes() *grafanaapiserver.APIRoutes {
+	tags := []string{"Create and Delete (custom routes)"}
+	return &grafanaapiserver.APIRoutes{
+		Namespace: []grafanaapiserver.APIRouteHandler{
+			{
+				Path: "dashsnaps/create",
+				Spec: &spec3.PathProps{
+					Summary:     "an example at the root level",
+					Description: "longer description here?",
+					Post: &spec3.Operation{
+						OperationProps: spec3.OperationProps{
+							Tags: tags,
+							RequestBody: &spec3.RequestBody{
+								RequestBodyProps: spec3.RequestBodyProps{
+									Description: "TODO???? can we get the request+response shapes here",
+								},
+							},
+						},
+					},
+				},
+				Handler: func(w http.ResponseWriter, r *http.Request) {
+					user, err := appcontext.User(r.Context())
+					if err != nil {
+						w.WriteHeader(500)
+						return
+					}
+					vars := mux.Vars(r)
+					info, err := request.ParseNamespace(vars["namespace"])
+					if err != nil {
+						_, _ = w.Write([]byte("expected namespace"))
+						w.WriteHeader(400)
+						return
+					}
+					if info.OrgID != user.OrgID {
+						_, _ = w.Write([]byte("org id mismatch"))
+						w.WriteHeader(401)
+						return
+					}
+					wrap := &contextmodel.ReqContext{
+						Logger: b.logger,
+						Context: &web.Context{
+							Req:  r,
+							Resp: web.NewResponseWriter(r.Method, w),
+						},
+						SignedInUser: user,
+					}
+					opts, err := b.options(info.Value)
+					if err != nil {
+						wrap.JsonApiErr(http.StatusBadRequest, "error getting options", err)
+						return
+					}
+
+					// This also writes the response
+					dashboardsnapshots.CreateDashboardSnapshot(wrap, opts.Spec, b.service)
+				},
+			},
+			{
+				Path: "dashsnaps/delete/{deleteKey}",
+				Spec: &spec3.PathProps{
+					Summary:     "an example at the root level",
+					Description: "longer description here?",
+					Delete: &spec3.Operation{
+						OperationProps: spec3.OperationProps{
+							Tags: tags,
+						},
+					},
+				},
+				Handler: func(w http.ResponseWriter, r *http.Request) {
+					ctx := r.Context()
+					vars := mux.Vars(r)
+					key := vars["deleteKey"]
+
+					err := dashboardsnapshots.DeleteWithKey(ctx, key, b.service)
+					if err != nil {
+						_, _ = w.Write([]byte("Failed to delete external dashboard"))
+						w.WriteHeader(500)
+						return
+					}
+
+					js, _ := json.Marshal(&util.DynMap{
+						"message": "Snapshot deleted. It might take an hour before it's cleared from any CDN caches.",
+					})
+					_, _ = w.Write(js)
+					w.WriteHeader(200)
+				},
+			},
+		},
+	}
+}
+
+func (b *SnapshotsAPIBuilder) GetAuthorizer() authorizer.Authorizer {
+	return authorizer.AuthorizerFunc(
+		func(ctx context.Context, attr authorizer.Attributes) (authorized authorizer.Decision, reason string, err error) {
+			// TODO -- something more restrictive?
+			return authorizer.DecisionAllow, "", err
```

Comment:
TODO... this is the biggest open question now.  We must audit that the existing permissions model still holds up.